File: frontend/test-supabase.js
// Simple Node.js script to test Supabase connection for frontend
// Run with: node test-supabase.js

require('dotenv').config();
const { createClient } = require('@supabase/supabase-js');

// Get Supabase credentials from environment variables
const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error('❌ Missing Supabase credentials. Make sure NEXT_PUBLIC_SUPABASE_URL and NEXT_PUBLIC_SUPABASE_ANON_KEY are set in .env file');
  process.exit(1);
}

console.log('Testing Supabase connection for frontend...');
console.log(`URL: ${supabaseUrl}`);
console.log(`Key: ${supabaseKey.substring(0, 10)}...${supabaseKey.substring(supabaseKey.length - 5)}`);

async function testConnection() {
  try {
    // Initialize Supabase client
    const supabase = createClient(supabaseUrl, supabaseKey);
    
    // Test 1: Check connection by querying public tables
    console.log('\nTest 1: Querying public tables...');
    const { data: tablesData, error: tablesError } = await supabase
      .from('user_profiles')
      .select('*')
      .limit(1);
    
    if (tablesError) {
      console.error(`❌ Error querying tables: ${tablesError.message}`);
    } else {
      console.log('✅ Successfully connected to Supabase!');
      console.log(`Retrieved data: ${JSON.stringify(tablesData, null, 2)}`);
    }
    
    // Test 2: Check authentication (anon access)
    console.log('\nTest 2: Checking auth status...');
    const { data: authData, error: authError } = await supabase.auth.getSession();
    
    if (authError) {
      console.error(`❌ Auth error: ${authError.message}`);
    } else {
      console.log('✅ Auth service working');
      console.log(`Auth status: ${authData.session ? 'Logged in' : 'Not logged in'}`);
    }
    
    // Test 3: Check storage access by trying to access the bucket directly
    console.log('\nTest 3: Checking storage access...');
    try {
      // Try to access the bucket directly by listing files
      const { data: files, error: filesError } = await supabase.storage
        .from('transcriptpro-files')
        .list();
      
      if (filesError) {
        console.error(`❌ Error accessing storage bucket: ${filesError.message}`);
      } else {
        console.log('✅ Storage bucket "transcriptpro-files" is accessible');
        console.log(`Found ${files.length} files in the bucket (empty is normal)`);
      }
    } catch (e) {
      console.error(`❌ Storage error: ${e.message}`);
    }
    
    console.log('\nTest completed!');
    
  } catch (error) {
    console.error(`❌ Connection failed: ${error.message}`);
  }
}

testConnection(); 

File: backend/test_supabase.py
import os
import asyncio
from dotenv import load_dotenv
from supabase import create_client

# Manually load environment variables from .env file
load_dotenv()

# Get Supabase credentials 
SUPABASE_URL = os.getenv("SUPABASE_URL")
SUPABASE_KEY = os.getenv("SUPABASE_SERVICE_KEY")

print(f"SUPABASE_URL: {SUPABASE_URL}")
print(f"SUPABASE_KEY: {SUPABASE_KEY[:10]}...{SUPABASE_KEY[-5:] if SUPABASE_KEY else ''}")

async def test_connection():
    """Test Supabase connection and basic operations"""
    print("\nTesting Supabase connection...")
    
    if not SUPABASE_URL or not SUPABASE_KEY:
        print("❌ Missing Supabase credentials. Please check your .env file.")
        return
    
    try:
        # Create Supabase client directly
        supabase = create_client(SUPABASE_URL, SUPABASE_KEY)
        
        # Test 1: Check if we can connect by retrieving user profiles
        print("Test 1: Querying user_profiles table...")
        setting_response = supabase.table("user_profiles").select("*").limit(1).execute()
        print(f"✅ Connection successful! Response data: {setting_response.data}")
        
        # Test 2: List tables in public schema
        print("\nTest 2: Listing tables in the public schema:")
        try:
            tables = supabase.rpc('get_tables').execute()
            if tables.data:
                for table in tables.data:
                    print(f"- {table}")
            else:
                print("No tables found or insufficient permissions.")
        except Exception as e:
            print(f"Cannot query tables: {str(e)}")
            # Fallback - just list the tables we know should exist
            print("Expected tables: user_profiles, files, transcriptions")
            
        # Test 3: Check storage bucket
        print("\nTest 3: Checking storage bucket...")
        try:
            bucket_list = supabase.storage.list_buckets()
            print(f"Available buckets: {[b['name'] for b in bucket_list]}")
            
            if any(b['name'] == 'transcriptpro-files' for b in bucket_list):
                print("✅ Storage bucket 'transcriptpro-files' exists")
            else:
                print("❌ Storage bucket 'transcriptpro-files' not found")
        except Exception as e:
            print(f"❌ Storage bucket error: {str(e)}")
        
        print("\nAll tests completed!")
        
    except Exception as e:
        print(f"❌ Connection failed: {str(e)}")

if __name__ == "__main__":
    asyncio.run(test_connection()) 